sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/AbstractSparkSQLParser.scala

```diff
@@ -25,15 +25,42 @@ import scala.util.parsing.input.CharArrayReader.EofCh
 
 import org.apache.spark.sql.catalyst.plans.logical._
 
+private[sql] object KeywordNormalizer {
```

Comment:
It's also used within`SqlLexical.processIdent`, but you're right, we'd better keep it the minimize visibility. I will do that in #4015 .